exercises/0002_eiger_handler.py

```diff
@@ -0,0 +1,102 @@
+#0002
+from databroker import Broker
+import pandas as pd
+import os
+
+import datetime
+import time
+from time import mktime
+
+from eiger_io.fs_handler import EigerHandler
+from databroker.assets.handlers import AreaDetectorTiffHandler
+
+def file_sizes(hdrs, db):
+
+    unique_resources = set()
+    time_size = dict()
+    file_sizes = list()
+    FILESTORE_KEY = "FILESTORE:"
+    start_time = time.time()
+    timestamp = 0.0
+    for hdr in hdrs:
+        for stream_name in hdr.stream_names:
+            events = hdr.events(stream_name=stream_name)
+            events = iter(events)
+            while True:
+                try:
+                    event = next(events)
+                    if "filled" in event:
+                        # there are keys that may not be filled
+                        for key, val in event['filled'].items():
+                            # if this is true, then we have a datum
+                            if not val:
+                                # get the datum
+                                if key in event['data']:
+                                    datum_id = event['data'][key]
+                                    resource = (db.reg.resource_given_datum_id(datum_id))
+                                    resource_id = resource['uid']
+                                    if resource_id in unique_resources:
+                                        continue
+                                    else:
+                                        unique_resources.add(resource_id)
+                                    datum_gen = db.reg.datum_gen_given_resource(resource_id)
+                                    datum_kwargs = [datum['datum_kwargs'] for datum in datum_gen]
+                                    timestamp = time.strftime('%Y-%m-%d %H:%M:%S', time.localtime(int(event['time'])))
+                                    timestamp = time.strptime(timestamp, '%Y-%m-%d %H:%M:%S')
+                                    timestamp = datetime.datetime.fromtimestamp(mktime(timestamp))
+                                    # get the file handler using this
+                                    fh = db.reg.get_spec_handler(resource_id)
+                                    print(fh)
+                                    file_sizes = sum(fh.get_file_sizes(datum_kwargs))
```

Comment:
call `get_file_list` here, get the file list and then get the file size from there.

Reply:
The file list that is returned includes more than one file, which means I would have to get the size for both files, sum them up, and then assign the sum of the file usage to the file size correct? So I'd need a loop in there?

Reply:
yep that's right. you can use list comprehension:
```py
filenames = fh.get_file_list(...)
total_size = sum([os.path.getsize(fliename) for filename in filenames])

```

Reply:
I think I got it, I'll push it right now

Reply:
Oh wow, I didn't see your message, but that's exactly what I did, thanks!

Reply:
awesome :-)